core/templates/pages/exploration-editor-page/editor-tab/training-panel/training-data-editor-panel-modal.controller.spec.ts

```diff
@@ -0,0 +1,464 @@
+            voiceovers_mapping: {
+              feedback_1: {}
+            },
+          },
+          param_changes: [],
+          interaction: {
+            answer_groups: [{
+              rule_specs: [],
+              outcome: {
+                dest: 'Hola',
+                feedback: {
+                  content_id: 'feedback_1',
+                  html: '',
+                },
+              },
+              training_data: ['Answer2']
+            }],
+            default_outcome: {
+              dest: 'Hola',
+              feedback: {
+                content_id: 'feedback_1',
+                html: '',
+              },
+            },
+            hints: [],
+            id: 'TextInput',
+            solution: null,
+          },
+          solicit_answer_details: false,
+          written_translations: {
+            translations_mapping: {
+              feedback_1: {}
+            },
+          },
+        },
+      });
+      ResponsesService.init(InteractionObjectFactory.createFromBackendDict({
+        id: 'TextInput',
+        answer_groups: [{
+          outcome: {
+            dest: '',
+            feedback: {
+              content_id: 'feedback_2',
+              html: ''
+            },
+          },
+          rule_specs: [{
+            type: '',
+            inputs: {
+              x: ['c', 'd', 'e'],
+              y: ['a', 'b', 'c']
+            }
+          }],
+          training_data: ['Answer1', 'Answer2']
+        }],
+        default_outcome: {
+          dest: 'Hola',
+          feedback: {
+            content_id: 'feedback_1',
+            html: '',
+          },
+        },
+        confirmed_unclassified_answers: [],
+        customization_args: {
+          rows: {
+            value: true
+          },
+          placeholder: {
+            value: 1
+          }
+        },
+        hints: [],
+      }));
+      ResponsesService.changeActiveAnswerGroupIndex(0);
+      StateInteractionIdService.init('Hola', 'TextInput');
+      StateCustomizationArgsService.init('Hola', {});
+
+      var $rootScope = $injector.get('$rootScope');
+
+      $uibModalInstance = jasmine.createSpyObj(
+        '$uibModalInstance', ['close', 'dismiss']);
+
+      $scope = $rootScope.$new();
+      $controller(
+        'TrainingDataEditorPanelServiceModalController', {
+          $scope: $scope,
+          $uibModalInstance: $uibModalInstance
+        });
+    }));
+
+    it('should evaluate scope variables values correctly', function() {
+      expect($scope.stateName).toBe('Hola');
+      expect($scope.stateContent).toBe('This is Hola State');
+      expect($scope.answerGroupHasNonEmptyRules).toBe(true);
+      expect($scope.inputTemplate).toBe(
+        '<oppia-interactive-text-input last-answer="null"' +
+        ' label-for-focus-target="testInteractionInput">' +
+        '</oppia-interactive-text-input>');
+    });
+
+    it('should call init when controller is initialized', function() {
```

Comment:
I see these patterns across controller tests. There is one test to make sure scope variables are correctly initialized. There is another to make sure that the init works as expected. Might be worth noting this in the wiki page.

Reply:
Sure! I'll put this in the wiki page!

Reply:
@marianazangrossi, did you do this yet?

Reply:
Not yet, I'm going to write it today!